Student/Student/Submissions/SubmissionButton/SubmissionButtonPresenter.swift

```diff
@@ -72,7 +72,7 @@ class SubmissionButtonPresenter: NSObject {
         }
         if quiz?.submission?.attemptsLeft == 0 { return nil }
         if assignment.quizID != nil {
-            return assignment.submission?.workflowState == .unsubmitted
+            return assignment.submission?.submittedAt == nil
```

Comment:
So we can't trust the workflow state of quiz assignments, is that right?

Reply:
It's not that we can't trust it, it's that there are multiple states the submission can be in when considering submission and graded. Since `workflowState` only tracks one possible state it's unreliable for this situation. If a user is unsubmitted but greaded, the workflow state is `graded` which still doesn't tell us the status of their submission.